 B. Perform stemming and lemmatization (40%) 
1. Use porter stemmer to run stemming. Count the number of unique words/tokens before and after stemming 

LLM used: ChatGPT
Prompts first: Write a python code in ipynb format for Perform stemming and lemmatization (40%), Use porter stemmer to run stemming. Count the number of unique words/tokens before and after stemming 
Prompt last: explain why nltk punkt library used

In [17]:
# Import necessary libraries

import pandas as pd

from nltk.tokenize import word_tokenize

from nltk.stem import PorterStemmer

from nltk.corpus import stopwords

import string

import nltk
import re
 
# Download required NLTK data for handling punctations

nltk.download('punkt')

# Import the corpus
corpus = pd.read_csv('C:\\PG AI and DS\\Semester 3\\NLP AISC2009\\Data.csv')

# Check the first 5 documents
corpus.head()
 
# Combine all text data into a single string (adjust the column name as needed)

text_data = " ".join(corpus.astype(str).apply(" ".join, axis=1))
 
# Tokenize the text

tokens = word_tokenize(text_data)
 
# Remove punctuation and make lowercase
tokens = [word.lower() for word in tokens if word.isalpha()]
 
# Count unique tokens before stemming

unique_tokens_before = set(tokens)

print(f"Number of unique tokens before stemming: {len(unique_tokens_before)}")
 
# Apply Porter Stemmer

stemmer = PorterStemmer()

stemmed_tokens = [stemmer.stem(word) for word in tokens]
 
# Count unique tokens after stemming

unique_tokens_after = set(stemmed_tokens)

print(f"Number of unique tokens after stemming: {len(unique_tokens_after)}")

# Optional: show some examples
for original, stemmed in zip(tokens[:10], stemmed_tokens[:10]):
    print(f"{original} → {stemmed}")


[nltk_data] Downloading package punkt to
[nltk_data]     C:\Users\Admin\AppData\Roaming\nltk_data...
[nltk_data]   Package punkt is already up-to-date!


Number of unique tokens before stemming: 7121
Number of unique tokens after stemming: 5822
watch → watch
or → or
listen → listen
live → live
weekdays → weekday
at → at
mt → mt
at → at
subscribe → subscrib
via → via


Result:
Number of unique tokens before stemming: 7121
Number of unique tokens after stemming: 5822
watch → watch
or → or
listen → listen
live → live
weekdays → weekday
at → at
mt → mt
at → at
subscribe → subscrib
via → via